Reflection/CustomConfigurationDemo/Attributes/ConfigurationItemAttribute.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CustomConfigurationDemo.Attributes;
+
+[AttributeUsage(AttributeTargets.Property)]
+public abstract class ConfigurationItemAttribute : Attribute
```

Comment:
It's a good practise to name abstract classes with `Base` prefix, it should look like `BaseConfigurationItemAttribute`